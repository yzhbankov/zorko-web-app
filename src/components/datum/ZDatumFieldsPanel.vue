<template>
  <v-card>
    <v-card-text>
      <v-subheader>Fields</v-subheader>
      <v-list>
        <v-list-tile
          avatar
          v-for="field in fields"
          :key="field.text">
          <v-list-tile-content>
            <v-list-tile-title>{{ field.text }}</v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
    </v-card-text>
  </v-card>
</template>

<script>
export default {
  name: 'ZDatumFieldsPanel',
  props: {
    fields: {
      type: Array
    }
  }
}
</script>
