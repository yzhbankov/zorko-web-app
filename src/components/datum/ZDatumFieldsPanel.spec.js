import {mount, shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import ZDatumFieldsPanel from './ZDatumFieldsPanel.vue'

describe('ZDatumFieldsPanel.vue', () => {
  let defaultProps

  beforeEach(() => {
    defaultProps = {
      fields: [
        {
          'text': 'Name'
        },
        {
          'text': 'Miles_per_Gallon'
        },
        {
          'text': 'Cylinders'
        }
      ]
    }
  })

  it('can initialize', () => {
    const wrapper = mount(ZDatumFieldsPanel, {
      propsData: defaultProps
    })

    expect(wrapper.vm).toBeTruthy()
  })

  it('has same HTML structure', () => {
    const renderer = createRenderer()
    const wrapper = shallow(ZDatumFieldsPanel, {
      propsData: defaultProps
    })

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
