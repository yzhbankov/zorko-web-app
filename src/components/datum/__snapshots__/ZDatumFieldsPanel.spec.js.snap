// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZDatumFieldsPanel.vue has same HTML structure 1`] = `
<v-card>
    <v-card-text>
        <v-subheader>Fields</v-subheader>
        <v-list>
            <v-list-tile avatar="">
                <v-list-tile-content>
                    <v-list-tile-title>Name</v-list-tile-title>
                </v-list-tile-content>
            </v-list-tile>
            <v-list-tile avatar="">
                <v-list-tile-content>
                    <v-list-tile-title>Miles_per_Gallon</v-list-tile-title>
                </v-list-tile-content>
            </v-list-tile>
            <v-list-tile avatar="">
                <v-list-tile-content>
                    <v-list-tile-title>Cylinders</v-list-tile-title>
                </v-list-tile-content>
            </v-list-tile>
        </v-list>
    </v-card-text>
</v-card>
`;
